src/behaviours/preferences/navigation-to-extension-specific-preferences.test.tsx

```diff
@@ -87,14 +147,202 @@ describe("preferences - navigation to extension specific preferences", () => {
 
           expect(actual).toBeNull();
         });
+
+        it("link is active", () => {
+          const actual = rendered.getByTestId("tab-link-for-extension-some-test-extension-id");
+
+          expect(actual).toHaveClass("active");
+        });
+      });
+    });
+
+    describe("given extension with registered tab", () => {
+      beforeEach(async () => {
+        const getRendererExtensionFake = getRendererExtensionFakeFor(applicationBuilder);
+        const extension = getRendererExtensionFake(extensionStubWithWithRegisteredTab);
+
+        await applicationBuilder.extensions.renderer.enable(extension);
+      });
+
+      it("shows extension tab in general area", () => {
+        const actual = rendered.getByTestId("tab-link-for-extension-registered-tab-page-id-1-nav-item-metrics-extension-tab");
+
+        expect(actual).toMatchSnapshot();
+      });
+
+      it("does not show custom settings block", () => {
+        const actual = rendered.queryByTestId("extension-settings");
+
+        expect(actual).not.toBeInTheDocument();
+      });
+
+      describe("when navigating to specific extension tab", () => {
+        beforeEach(() => {
+          applicationBuilder.preferences.navigation.click("extension-registered-tab-page-id-1-nav-item-metrics-extension-tab");
+        });
+        it("renders", () => {
+          expect(rendered.container).toMatchSnapshot();
+        });
+        it("shows related preferences for this tab", () => {
+          const actual = rendered.getByTestId("metrics-preference-item-hint");
+
+          expect(actual).toBeInTheDocument();
+        });
+        it("does not show unrelated preferences for this tab", () => {
+          const actual = rendered.queryByTestId("survey-preference-item-hint");
+
+          expect(actual).not.toBeInTheDocument();
+        });
+      });
+    });
+
+    describe("given extension with few registered tabs", () => {
+      beforeEach(async () => {
+        const getRendererExtensionFake = getRendererExtensionFakeFor(applicationBuilder);
+        const extension = getRendererExtensionFake(extensionStubWithWithRegisteredTabs);
+
+        await applicationBuilder.extensions.renderer.enable(extension);
+      });
+
+      it("shows each of registered tabs in general area", () => {
+        const helloTab = rendered.getByTestId("tab-link-for-extension-hello-world-tab-page-id-1-nav-item-hello-extension-tab");
+        const logsTab = rendered.getByTestId("tab-link-for-extension-hello-world-tab-page-id-1-nav-item-logs-extension-tab");
+
+        expect(helloTab).toBeInTheDocument();
+        expect(logsTab).toBeInTheDocument();
```

Comment:
These should be split to two separate tests for documentation and better error message in case of failure.